server/subscribe.go

```diff
@@ -206,10 +206,13 @@ func (p *Plugin) getChannelsSubscribed(wh *webhook, instanceID types.ID) ([]Chan
 	}
 
 	var channelSubscriptions []ChannelSubscription
+	subscriptionMap := make(map[string]bool)
 	subIds := subs.Channel.ByID
 	for _, sub := range subIds {
 		if p.matchesSubsciptionFilters(wh, sub.Filters) {
-			channelSubscriptions = append(channelSubscriptions, sub)
+			if !subscriptionMap[sub.ChannelID] {
+				channelSubscriptions = append(channelSubscriptions, sub)
```

Comment:
You need to add it to the map for it to avoid collisions.

Reply:
@wiggin77 updated the map for every iteration. Please have a look